tests/model_explainability/trustyai_service/utils.py

```diff
@@ -68,3 +74,69 @@ def _get_mariadb_pods() -> list[Pod]:
             condition=Pod.Condition.READY,
             status="True",
         )
+
+
+@retry(wait_timeout=Timeout.TIMEOUT_2MIN, sleep=5)
+def validate_trustyai_service_db_conn_failure(
+    client: DynamicClient, namespace: str, label_selector: Optional[str]
+) -> bool:
+    """
+    Waits for TrustyAIService pod to fail and checks if the pod is:-
+    * in a CrashLoopBackOff state
+    * The LastState is in terminated state and the cause was a MariaDB TLS certificate exception
+    Also checks if there are more than one pod for the service.
+
+    Args:
+        client: DynamicClient
+        namespace: Namespace
+        label_selector: Optional(str)
+
+    Returns: bool
+    """
+    pods = list(Pod.get(dyn_client=client, namespace=namespace, label_selector=label_selector))
+    if len(pods) != 0:
+        if len(pods) > 1:
+            pytest.fail("More than one pod found in TrustyAIService.")
+        for container_status in pods[0].instance.status.containerStatuses:
+            if (terminate_state := container_status.lastState.terminated) and terminate_state.reason in (
+                pods[0].Status.ERROR,
+                pods[0].Status.CRASH_LOOPBACK_OFF,
+            ):
+                if (
+                    "Could not connect to mariadb:" not in terminate_state.message
+                    and "java.security.cert.CertPathValidatorException: signature check failed"
+                    not in terminate_state.message
+                ):
```

Comment:
please use a regex to check `terminate_state.message` instead of 2 separate checks